src/java/htsjdk/samtools/SAMRecordDuplicateComparator.java

```diff
@@ -39,6 +39,10 @@
  * @author nhomer
  */
 public class SAMRecordDuplicateComparator implements SAMRecordComparator {
+    /** An enum to provide type-safe keys for transient attributes the comparator puts on SAMRecords. */
+    private static enum Attr {
+        LibraryId, ReadCoordinate, MateCoordinate
```

Comment:
Perhaps these useful attributes should be defined somewhere, like in ```SAMRecordUtils```? Your call as we have not needed it in the past.